hs_rest_api/resources/file_metadata.py

```diff
@@ -0,0 +1,184 @@
+                "east": -84.0465,
+                "north": 49.6791,
+                "name": "12232",
+                "projection": "WGS 84 EPSG:4326"
+            },
+            "extra_metadata": {
+                "extended1": "one"
+            },
+            "temporal_coverage": {
+                "start": "2018-02-22",
+                "end": "2018-02-24"
+            },
+            "title": "File Metadata Title"
+        }
+        ```
+        """
+        file_serializer = FileMetaDataSerializer(request.data)
+
+        title = file_serializer.data.pop("title", "")
+        resource_file = ResourceFile.objects.get(id=file_id)
+        resource_file.metadata.logical_file.dataset_name = title
+        resource_file.metadata.logical_file.save()
+
+        spatial_coverage = file_serializer.data.pop("spatial_coverage", None)
+        if spatial_coverage is not None:
+            if resource_file.metadata.spatial_coverage is not None:
+                Coverage.update(resource_file.metadata.spatial_coverage.id,
+                                _value=json.dumps(spatial_coverage))
+            elif resource_file.metadata.spatial_coverage is None:
+                resource_file.metadata.create_element('coverage', type="point",
+                                                      _value=json.dumps(spatial_coverage))
+
+        temporal_coverage = file_serializer.data.pop("temporal_coverage", None)
+        if temporal_coverage is not None:
+            if resource_file.metadata.temporal_coverage is not None:
+                Coverage.update(resource_file.metadata.temporal_coverage.id,
+                                _value=json.dumps(temporal_coverage))
+            elif resource_file.metadata.temporal_coverage is None:
+                resource_file.metadata.create_element('coverage', type="period",
+                                                      _value=json.dumps(temporal_coverage))
+
+        keywords = file_serializer.data.pop("keywords", None)
+        if keywords is not None:
+            resource_file.metadata.keywords = keywords
+
+        extra_metadata = file_serializer.data.pop("extra_metadata", None)
+        if extra_metadata is not None:
+            resource_file.metadata.extra_metadata = extra_metadata
+
+        resource_file.metadata.save()
+
+        # TODO: How to leverage serializer for this?
+        title = resource_file.metadata.logical_file.dataset_name \
+            if resource_file.metadata.logical_file else ""
+        keywords = resource_file.metadata.keywords \
+            if resource_file.metadata else []
+        spatial_coverage = resource_file.metadata.spatial_coverage.value \
+            if resource_file.metadata.spatial_coverage else {}
+        extra_metadata = resource_file.metadata.extra_metadata \
+            if resource_file.metadata else []
+        temporal_coverage = resource_file.metadata.temporal_coverage.value if \
+            resource_file.metadata.temporal_coverage else {}
+
```

Comment:
I think instead of having all these code logic here if we can make that as part of a function of the FileMetaData class similar to Resource MetaData class then later each File type metadata class can override the logic to meet  it's own metadata editing requirements. Here is the reference to the  use of a similar function in the case of updating resource level metadata in an endpoint:

https://github.com/hydroshare/hydroshare/blob/develop/hs_core/views/resource_metadata_rest_api.py#L196

Reply:
I agree, though I think this is beyond the scope of this body of work which is specific to one use case.

I've made a ticket here: https://github.com/hydroshare/hydroshare/issues/2639

Reply:
@aphelionz Sounds good.